File: src/main/java/no/novari/flyt/history/model/action/ManuallyProcessedEventAction.java
package no.novari.flyt.history.model.action;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.extern.jackson.Jacksonized;
import no.novari.flyt.history.model.SourceApplicationAggregateInstanceId;

@Jacksonized
@Builder
@Getter
@Setter
public class ManuallyProcessedEventAction implements SourceApplicationAggregateInstanceId {
    @NotNull
    private Long sourceApplicationId;
    @NotEmpty
    private String sourceApplicationIntegrationId;
    @NotEmpty
    private String sourceApplicationInstanceId;
    @NotEmpty
    private String archiveInstanceId;
}

File: src/main/kotlin/no/novari/flyt/history/model/action/ManuallyProcessedEventAction.kt
package no.novari.flyt.history.model.action

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Positive
import no.novari.flyt.history.model.SourceApplicationAggregateInstanceId

@Schema(description = "Request to mark an instance as manually processed.")
data class ManuallyProcessedEventAction(
    @field:Positive
    override val sourceApplicationId: Long = 0,
    @field:NotEmpty
    override val sourceApplicationIntegrationId: String = "",
    @field:NotEmpty
    override val sourceApplicationInstanceId: String = "",
    @field:NotEmpty
    val archiveInstanceId: String = "",
) : SourceApplicationAggregateInstanceId
